docs/experiment-tracking.md

```diff
@@ -0,0 +1,134 @@
+
+Tracking can be enabled by passing a config in the [Training Arguments](https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/tuning/config/configs.py#L131)
+with the name of the enabled trackers passed as a list.
+
+```
+from tuning import sft_trainer
+
+training_args = {
+    ..
+    ..
+    trackers: ["aim", "file_logger"]
+}
+
+sft_trainer.train(train_args=training_args,....)
+```
+
+For each of the requested trackers the code expects you to pass a config to the `sft_trainer.train` function as well which can be specified as the `tracker_conifgs` argument [here](https://github.com/foundation-model-stack/fms-hf-tuning/blob/main/tuning/sft_trainer.py#L76) details of which are present below.
+
+
+The list of available trackers and their details are as follows,
+
+## Tracker Configs.
+
+## File Logging Tracker
+
+[File Logger](../tuning/trackers/filelogging_tracker.py) is an inbuilt tracker which can be used to dump contents of the loss curve generated while training to a file.
+
+Currently File Logger is enabled by default and will dump the loss curve of a training to a default file path specified [here](../tuning/config/tracker_configs.py) inside the output folder passed during training.
+
+To override the location of file logger please pass an instance of the [FileLoggingTrackerConfig](../tuning/config/tracker_configs.py) in the `tracker_configs` argument.
+
+```
+from tuning import sft_trainer
+from tuning.config.tracker_configs import FileLoggingTrackerConfig, TrackerConfigFactory
+
+training_args = {
+    ..
+    ..
+    trackers: ["file_logger"]
+}
+
+
+logs_file = "new_train_logs.jsonl"
+
+tracker_configs = TrackerConfigFactory(
+    file_logger_config=FileLoggingTrackerConfig(
+        training_logs_filename=logs_file
+        )
+    )
+
+sft_trainer.train(train_args=training_args, tracker_configs=tracker_configs,....)
+```
+
+Currently File Logging tacker supports only one argument and this file will be placed inside the `train_args.output` folder.
+
+## Aimstack Tracker
+
+To enable [Aim](https://aimstack.io) users need to pass `"aim"` as the requested tracker.
+
+
+When using Aimstack, users need to specify additional arguments which specify where the Aimstack database is present and what experiment name to use
+for tracking the training.
+
+Aimstack supports either a local (`filesystem path`) based db location or a remote (`aim_server:port`) based databse location.
+See Aim [documentation](https://aimstack.readthedocs.io/en/latest/using/remote_tracking.html) for more details.
+
+After [initialising a repo](https://aimstack.readthedocs.io/en/latest/quick_start/setup.html#initializing-aim-repository), users can specify the location of the
```

Comment:
Should we also provide commands to initialise a aim repo, what you think?

Reply:
lets leave that to the aim documentation..it is aim specific.